### DISCUSSION TITLE
Whole page gets re-mounted after a locale change?

### DISCUSSION THREAD

**Discussion by nikola1970** [amannn/next-intl#723] [Q&A]
Hello, so everything looks to work fine besides that whenever I switch locale whole page gets re-mounted (very short blank flash) for some reason.

I am consuming all my translations `client side`.

App structure is like this:
`app/[locale]/layout.tsx` etc, so all pages are nested inside locale folder.

My `[locale]/layout.tsx` looks like this:

```tsx
export default function RootLayout({
  children,
  params: { locale = 'de' },
}: {
  children: React.ReactNode;
  params: { locale: string };
}) {
  const messages = useMessages();

  if (!LOCALES.includes(locale as any)) {
    notFound();
  }

  return (
    <html lang={locale}>
      <body>
        <NextIntlClientProvider messages={messages}>
          <RootProviders>{children}</RootProviders>
        </NextIntlClientProvider>
      </body>
    </html>
  );
}
```

My navigation utils file is this:

```tsx
export const { Link, redirect, useRouter } = createSharedPathnamesNavigation({
  locales: LOCALES,
});
```

And middleware is like this:

```ts
export default createMiddleware({
  locales: LOCALES,
  defaultLocale: 'de',
});

export const config = {
  // Skip all paths that should not be internationalized. This example skips
  // certain folders and all pathnames with a dot (e.g. favicon.ico)
  matcher: ['/((?!api|_next|_vercel|.*\\..*).*)'],
};

```

When using the link provided by `createSharedPathnamesNavigation`, it triggers a full-page reload. Because of this, I utilize the next/navigation link with appropriate linking to switch the language, and that works fine without hard-refreshing the page.

The only issue I encounter is that the page gets re-mounted upon any locale switch. For instance, if I have a form with input fields, and some of them are filled, switching the locale causes everything to be lost because the entire page is being remounted.

Any ideas? :) Thanks a lot!

**Comment by amannn** [comment]
Next.js remounting the page on locale change is expected since this is essentially a navigation to another page (see e.g. https://github.com/vercel/next.js/issues/44793). 

If you want to persist some state between navigation, you need to move it out of the local state of a component that gets remounted (e.g. to a context value that is rendered from the root layout).

**Reply by niksimFS24** [reply]
Thanks for the feedback. Will take a look :)
